bigquery/google/cloud/bigquery/table.py

```diff
@@ -1099,3 +1148,109 @@ def to_dataframe(self):
         rows = [row.values() for row in iter(self)]
 
         return pandas.DataFrame(rows, columns=column_headers)
+
+
+class TimePartitioningType(object):
+    """Specifies the type of time partitioning to perform."""
+
+    DAY = 'DAY'
+    """str: Generates one partition per day."""
+
+
+class TimePartitioning(object):
```

Comment:
Could you also add `requirePartitionFilter` while you're here (currently beta feature)? Search for it in https://www.googleapis.com/discovery/v1/apis/bigquery/v2/rest